controller/registries/consul/config.go

```diff
@@ -160,13 +168,28 @@ func (reg *Consul) Reload(c registrytype.RegistryConfig) error {
 	return nil
 }
 
-func (reg *Consul) refresh() error {
-	return nil
-}
+func (reg *Consul) fetchAllServices(client *Client) map[consulService]bool {
+	q := &consulapi.QueryOptions{}
+	q.Datacenter = client.DataCenter
+	q.Namespace = client.NameSpace
+	q.Token = client.Token
+	services, _, err := client.consulCatalog.Services(q)
 
-func (reg *Consul) fetchAllServices(client *Client) (map[consulService]bool, error) {
-	fmt.Println(client)
-	return nil, nil
+	if err != nil {
+		reg.logger.Errorf("failed to get service, err: %v", err)
+		return nil
```

Comment:
IMHO, we can pass the err out and make the Start fail if we fail to fetchAllServices. This can detect error like configuring a wrong consul address.